File: solutions/lua/resistor-color-duo/1/resistor-color-duo.lua
local color_map = {
  black = 0,
  brown = 1,
  red = 2,
  orange = 3,
  yellow = 4,
  green = 5,
  blue = 6,
  violet = 7,
  grey = 8,
  white = 9,
}
return {
  value = function(colors)
    return color_map[colors[1]] * 10 + color_map[colors[2]]
  end
}

File: solutions/ruby/resistor-color-duo/1/resistor_color_duo.rb
# frozen_string_literal: true

# ResistorColorDuo
module ResistorColorDuo
  def self.value(colors)
    h = Hash[%w[black brown red orange yellow green blue violet grey white].zip(0..)]
    h[colors[0]]*10 + h[colors[1]]
  end
end
